### PR TITLE
Tests: Add tests for Create Context publish attribute saves

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Add tests for Create Context saves and changes for `CreatedInstance.publish_attributes`

## Additional info

Test for fix: https://github.com/ynput/OpenPype/pull/5809

**Note:** I couldn't find enough information on how to define the tests. It will currently fail because it cannot find the test project, test asset and test task - but I have no idea how to make it so that `ModuleUnitTest` sets up just that for me without requiring to point to a download URL that I have no control over or have no knowledge about what to point it to. So currently it will fail but I don't know how to proceed.

## Testing notes:

1. Run test, it should pass.

### CODE REVIEW

**Comment by BigRoy** [comment]
@iLLiCiTiT I've prototyped some tests here - but unfortunately I have no idea how to make sure the test database has the `test_project`, `test_asset` and `test_task`. I had assumed there would've been an easy existing entry point for that in the unit tests but I could not find it.

@kalisp any pointers?

Preferably there's just a fixture that allows me to `create_current_project_asset_task` or alike to ensure its existence.

File: tests/resources/test_data.zip
Comment (BigRoy) [inline tests/resources/test_data.zip:1]: What changes here @kalisp ? Doesn't seem related to the `clean_db` tests?

Comment (kalisp) [reply tests/resources/test_data.zip:1]: I copied same `dumps` as in `clean_db` for a good measure as this zip file is mentioned in readme. (There were unnecessary openpype logs etc.)

**Comment by kalisp** [comment]
@BigRoy could you please take a look at falling asserts if they are issue of my fixes or something else to move this PR forward?

**Comment by kalisp** [comment]
Ping @BigRoy .

**Comment by mkolar** [comment]
Because we're splitting OpenPype into ayon-core and individual host addons, this PR would have to be re-created to target one of those.

We're closing it down, but we'll he happy for a new PR to [ynput/ayon-core](https://github.com/ynput/ayon-core) or the host addon repository once it's up.
